### PR TITLE
Lazily load user list in channels on init, keep autocompletion sort on server

### PR DESCRIPTION
**Pull request by xPaw**
Less bandwidth and processing on initial page load.

### CODE REVIEW

File: src/models/chan.js [hunk 1]
```diff
@@ -95,6 +95,7 @@ Chan.prototype.getMode = function(name) {
 
 Chan.prototype.toJSON = function() {
 	var clone = _.clone(this);
+	clone.users = [];
```
Comment (YaManicKill) [inline src/models/chan.js]: Maybe a quick comment here just mentioning why we do this. I can see it would be confusing for people coming to it.

**Comment by xPaw** [comment]
Don't merge it, it breaks auto completion (sorting) as users will be empty.

**Comment by xPaw** [comment]
I have fixed auto completion!

File: client/js/render.js [hunk 2]
```diff
@@ -145,22 +145,10 @@ function renderChannelMessages(data) {
...
-
-	for (const i in data.users) {
-		nicks.push(data.users[i].nick);
-	}
-
-	nicks = nicks.sort(function(a, b) {
-		return (oldSortOrder[a] || Number.MAX_VALUE) - (oldSortOrder[b] || Number.MAX_VALUE);
-	});
+	const nicks = data.users
+		.concat()
```
Comment (YaManicKill) [inline client/js/render.js]: This does nothing, unless I'm missing something.

File: client/js/render.js [hunk 3]
```diff
@@ -145,22 +145,10 @@ function renderChannelMessages(data) {
...
+		.sort((a, b) => b.lastMessage - a.lastMessage)
```
Comment (YaManicKill) [inline client/js/render.js]: I think I must have missed the point of renderChannelUsers. Is it not for the users for the sidebar? Because it looks like we are sorting based on when they last sent a message, which wouldn't make sense for the sidebar. So is this for the autocompletion?

File: src/models/chan.js (hunk 1 above)
Comment (YaManicKill) [inline src/models/chan.js]: Yeah, still want this comment.

File: client/js/render.js (hunk 2 above)
Comment (xPaw) [inline client/js/render.js]: Make a copy of the object, because it sorts in-place.

File: client/js/render.js (hunk 3 above)
Comment (xPaw) [inline client/js/render.js]: This is sorting auto completion, yes.

File: client/js/render.js (hunk 2 above)
Comment (YaManicKill) [inline client/js/render.js]: Oh. Well that's a very confusing way of doing it. I thin kthe standard way to do it is with slice(0), isn't it?

Comment (xPaw) [inline client/js/render.js]: There's no difference. i'll add a comment here

**Comment by xPaw** [comment]
Fixed a case where last message time would be reset when NAMES event is received. This change should also improve GC performance as we will keep `User` objects instead of creating new ones for every user.

File: client/js/render.js [hunk 4]
```diff
@@ -83,7 +83,7 @@ function buildChatMessage(data) {
 		text.html(templates.actions[type](data.msg));
 	}
 
-	if ((type === "message" || type === "action") && chan.hasClass("channel")) {
+	if ((type === "message" || type === "action" || type === "notice") && chan.hasClass("channel")) {
```
Comment (astorije) [inline client/js/render.js]: Why did you need this addition? Sorry, not questioning, I'm just not seeing it 😅.

Comment (xPaw) [inline client/js/render.js]: Notices are processed in same callback on the server as normal messages and actions. So this change was needed to match ordering correctly.

**Comment by xPaw** [comment]
@astorije Commits are better left without squashing.
